.github/scripts/verify-commits.jq

```diff
@@ -0,0 +1,23 @@
+# Expects the list of commits on a pull requests as input.
+# See: https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request
+[
+  # Iterate over all commits
+  .[] |
+  # select maintainer commits
+  select(
+    .author.login == "JanMa" or
```

Comment:
I think this doesn't solve this concern: https://github.com/openbao/openbao/issues/399#issuecomment-2267192422

In particular, I think we could perhaps more strongly enforce a three-fold condition here:

1. PRs should only have one login for a series of commits (additional authors should be moved to `Co-authored-by` tags, though note a particular user may have multiple emails associated with their GH account).
2. A PR's author should match the author of the commits.
3. All commits from maintainers should be signed.

This prevents the concern articulated there, but uniformly, for everyone. :-)

Reply:
I think my jq-foo is weak, but perhaps:

```jq
# Expects the list of commits on a pull requests as input.
# See: https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request
[
 # Iterate over all commits
 .[] |
 # Select commit authors
 .author.login
] |
# Check if there are multiple authors
if (. | length) != 1 then
 # Return error
 ("PR contains commits from multiple GitHub users!\n" | halt_error(1))
elif .[0] != "GITHUB_PR_AUTHOR" then
 # Return error
 ("PR contains commits from other than the PR author!\n" | halt_error(1))
else
 # Return success
 "All maintainer commits are signed"
end
```

(and then, add another call to `gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/openbao/openbao/pulls/447 | jq -r .user.login` and `sed` `GITHUB_PR_AUTHOR` to the desired value from this call?)

This would be in addition to the current script and enforce 1 & 2

Reply:
I think I disagree that points 1 and 2 should be implemented. It is common enough that two people collaborate on a PR and both push commits to the branch it's being worked on and we shouldn't prohibit this.

I've also seen maintainers support community members with their PRs by pushing commits to the PR branch, and when opening a PR you can explicitly opt-in to allowing this. By enforcing those rules we block ourselves from doing so in the future, and helping community members get their changes merged.

Reply:
Hmm, if the maintainer or anyone else is pushing commits, the result will be squashed anyways so there'll still only be one author with other commit authors moved to `Co-authored-by`. So having them set an explicit `--author=<PR Owner>` and then add a `Co-authored-by <themselves>` would end up being equivalent tbh, though admittedly more work.

But I do agree it is useful; usually I'd just take over community PRs at HashiCorp if the author had disappeared (but signed the CLA) and we wanted to merge things in  a more timely manner. 

We can skip this, but I'd still like to see some guard preventing others from pushing (even signed) commits to PRs. Otherwise, someone could take signed commits from our forks and push them in a PR (these would need to be taken verbatim, and their modifications added in subsequent commits).